src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -243,7 +247,7 @@ func (r *StatefulSetResource) Obj() (k8sclient.Object, error) {
 					ObjectMeta: metav1.ObjectMeta{
 						Namespace:	r.pandaCluster.Namespace,
 						Name:		"datadir",
-						Labels:		r.pandaCluster.Labels,
+						Labels:		clusterLabels,
```

Comment:
nit: I don't usually see any labels in VolumeClaimTemplates but haven't found a recommendation..

Reply:
hmm good point 🤔 I guess we can add those later as well, I'll note it down